CoreFoundation/Parsing.subproj/CFBinaryPList.c

```diff
@@ -501,7 +501,7 @@ static void _flattenPlist(CFPropertyListRef plist, CFMutableArrayRef objlist, CF
     CFDictionaryAddValue(objtable, plist, (const void *)(uintptr_t)refnum);
     if (_kCFRuntimeIDCFDictionary == type) {
         CFIndex count = CFDictionaryGetCount((CFDictionaryRef)plist);
-        STACK_BUFFER_DECL(CFPropertyListRef, buffer, count <= 128 ? count * 2 : 1);
+        STACK_BUFFER_DECL(CFPropertyListRef, buffer, (count > 0 && count <= 128) ? count * 2 : 1);
```

Comment:
Does this same check need to be repeated on the next line?

Reply:
It shouldn't be needed because if `count == 0` then the stack allocation is used instead of the `malloc()`'d buffer. 1 byte is wasted on the stack but this is also the case if `count > 128` and the heap is used instead of the stack.